src/editDistance.ts

```diff
@@ -111,4 +147,60 @@ function generateErrorItemArray(
   return errorList;
 }
 
-main('nick', 'nwck');
+function generateErrorGroupArray(errorItemArray: Array<ErrorItem>) {
+  return errorItemArray.reduceRight<ErrorGroup[]>((acc, cur, index) => {
```

Comment:
I forgot about reduce right... I'm wondering if we should try to build the ErrorItemArray using reduceRight instead of recursion. What do you think?